packages/netlify/src/index.ts

```diff
@@ -300,7 +301,11 @@ export default function netlifyIntegration(
 				logger.info('Emitted _redirects');
 
 				if (_config.output !== 'static') {
-					await writeSSRFunction();
+					let notFoundContent = undefined;
+					try {
+						notFoundContent = await readFile(new URL('./404.html', dir), 'utf8');
```

Comment:
filesystem check would work reliably. That said, do you think we should prefer checking the routes object?

https://github.com/withastro/astro/pull/9648/files#diff-071ccfbdfd7b9bc803219179f88ae0bdb227a399bc4a481b7831c82b801d7a17R298

Reply:
I am happy as long as the users get an improvement, which this PR already provides. Just meant to point out that it is an option.

As for preferences, I am of the opinion it should be upto the more active contributor, which would be you.

Reply:
I don't have any opinion here, fine with both approaches.